indico/modules/rb/controllers/backend/bookings.py

```diff
@@ -353,3 +362,38 @@ def _process(self):
         elif self.action == 'cancel':
             self.occurrence.cancel(session.user)
         return jsonify(occurrence=reservation_occurrences_schema.dump(self.occurrence, many=False))
+
+
+class RHBookingExport(RHRoomBookingBase):
+    @use_kwargs({
+        'room_ids': fields.List(fields.Int(), required=True),
+        'start_date': fields.Date(required=True),
+        'end_date': fields.Date(required=True),
+        'format': fields.Str(validate.OneOf({'csv', 'xlsx'}), required=True),
+    })
+    def _process(self, room_ids, start_date, end_date, format):
+        occurrences = (ReservationOccurrence.query
+                       .join(ReservationOccurrence.reservation)
+                       .filter(Reservation.room_id.in_(room_ids),
+                               ReservationOccurrence.is_valid,
+                               db_dates_overlap(ReservationOccurrence,
+                                                'start_dt', datetime.combine(start_date, time()),
+                                                'end_dt', datetime.combine(end_date, time.max)))).all()
+
+        token = unicode(uuid.uuid4())
+        headers, rows = generate_spreadsheet_from_occurrences(occurrences)
+        _cache.set(token, {'headers': headers, 'rows': rows}, time=1800)
+        download_url = url_for('rb.export_bookings_{}'.format(format), token=token)
```

Comment:
Please don't use dynamic endpoint names; that's awful when searching for the place where an endpoint is used.
Maybe use `bookings.<any(csv,xlsx):format>` in the route and handle both in a single RH? Then you can pass it as a kwarg to `url_for` here.